pkg/glusterutils/exporterd.go

```diff
@@ -63,28 +40,73 @@ func IsLeader(config *Config) (bool, error) {
 	}
 	return false, nil
 }
+func (g *GD2) IsLeader() (bool, error) {
+	peerList, err := g.Peers()
+	if err != nil {
+		return false, err
+	}
+	peerID, err := g.LocalPeerID()
+	if err != nil {
+		return false, err
+	}
+	for _, pr := range peerList {
+		if pr.Online {
+			if peerID == peerList[0].ID {
+				return true, nil
+			}
+			return false, nil
+		}
+	}
+	// This would imply none of the peers are online and return false
+	return false, nil
+}
 
-// VolumeInfo gets Volume info from Gd1/Gd2 based on config provided
-func VolumeInfo(config *Config) ([]Volume, error) {
+func MakeGluster(config *Config) ginterface {
 	setDefaultConfig(config)
-
-	if config.GlusterMgmt == MgmtGlusterd2 {
-		return gd2VolumeInfo(config)
+	if config.GlusterMgmt == "" || config.GlusterMgmt == MgmtGlusterd {
+		return &GD1{config: config}
+	} else {
+		return &GD2{config: config}
 	}
-
-	return gd1VolumeInfo(config)
 }
 
 // LocalPeerID returns local peer ID of glusterd/glusterd2
-func LocalPeerID(config *Config) (string, error) {
-	setDefaultConfig(config)
-
+func (g *GD1) LocalPeerID() (string, error) {
 	keywordID := "UUID"
-	peeridFile := config.GlusterdWorkdir + "/glusterd.info"
-	if config.GlusterMgmt == MgmtGlusterd2 {
-		keywordID = "peer-id"
-		peeridFile = config.GlusterdWorkdir + "/uuid.toml"
+	peeridFile := g.config.GlusterdWorkdir + "/glusterd.info"
+	fileStream, err := os.Open(filepath.Clean(peeridFile))
+	if err != nil {
+		return "", err
 	}
+	defer func() {
```

Comment:
below code is duplicated. create one more internal function and call from both `GD1.LocalPeerID` and `GD2.LocalPeerID`

Reply:
ack